src/core/server/integration_tests/saved_objects/migrations/group3/deferred_migrations.test.ts

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { TestElasticsearchUtils } from '@kbn/core-test-helpers-kbn-server';
+import type {
+  SavedObjectsRawDocSource,
+  SavedObjectsType,
+  SavedObjectUnsanitizedDoc,
+} from '@kbn/core-saved-objects-server';
+import { delay } from '../test_utils';
+
+import '../jest_matchers';
+import {
+  clearLog,
+  defaultKibanaIndex,
+  startElasticsearch,
+  KibanaMigratorTestKit,
+  getKibanaMigratorTestKit,
+} from '../kibana_migrator_test_kit';
+
+describe('deferred migrations', () => {
+  let client: KibanaMigratorTestKit['client'];
+  let runMigrations: KibanaMigratorTestKit['runMigrations'];
+  let savedObjectsRepository: KibanaMigratorTestKit['savedObjectsRepository'];
+  let server: TestElasticsearchUtils['es'];
+  let type: SavedObjectsType;
+
+  beforeAll(async () => {
+    server = await startElasticsearch();
+  });
+
+  afterAll(async () => {
+    await server?.stop();
+    await delay(10);
+  });
+
+  beforeEach(async () => {
+    const noop = (doc: SavedObjectUnsanitizedDoc) => doc;
+
+    type = {
+      name: 'some-type',
+      hidden: false,
+      namespaceType: 'agnostic',
+      mappings: {
+        properties: {
+          name: { type: 'keyword' },
+        },
+      },
+      migrations: {
+        '1.0.0': jest.fn(noop),
+        '2.0.0': jest.fn(noop),
+        '3.0.0': {
+          deferred: true,
+          transform: jest.fn(noop),
+        },
+        '4.0.0': jest.fn(noop),
+        '5.0.0': {
+          deferred: true,
+          transform: jest.fn(noop),
+        },
+        '6.0.0': {
+          deferred: true,
+          transform: jest.fn(noop),
+        },
+      },
+    };
+
+    ({ client, runMigrations, savedObjectsRepository } = await getKibanaMigratorTestKit({
+      kibanaIndex: defaultKibanaIndex,
+      types: [type],
+    }));
+    await clearLog();
+  });
+
+  describe.each`
+    source     | expected
+    ${'1.0.0'} | ${'6.0.0'}
+    ${'2.0.0'} | ${'6.0.0'}
+    ${'3.0.0'} | ${'6.0.0'}
+    ${'4.0.0'} | ${'4.0.0'}
+    ${'5.0.0'} | ${'5.0.0'}
+    ${'6.0.0'} | ${'6.0.0'}
```

Comment:
I'm probably missing something but I don't get this matrix. I would expect `1.0.0`, `2.0.0` and `3.0.0` to be migrated (have the persisted version) to `4.0.0`, not `6.0.0`?